WordPress/WordPressUITests/Utils/XCTest+Extensions.swift

```diff
@@ -2,29 +2,34 @@ import XCTest
 
 extension XCUIElement {
     /**
-     Removes any current text in the field before typing in the new value
-     - Parameter text: the text to enter into the field
+     Removes any current text in the field
      */
-    func clearAndEnterText(text: String) -> Void {
+    func clearTextIfNeeded() -> Void {
         let app = XCUIApplication()
         let content = self.value as! String
 
         if content.count > 0 && content != self.placeholderValue {
             self.press(forDuration: 1.2)
             app.menuItems["Select All"].tap()
-        } else {
-            self.tap()
+            app.menuItems["Cut"].tap()
         }
+    }
 
+    /**
+     Removes any current text in the field before typing in the new value
+     - Parameter text: the text to enter into the field
+     */
+    func clearAndEnterText(text: String) -> Void {
+        clearTextIfNeeded()
+        self.tap()
         self.typeText(text)
     }
 }
 
 extension XCTest {
 
     func isIPhone() -> Bool {
```

Comment:
@rachelmcr something I meant to mention in the last PR but it slipped me. It seems a bit weird for the `isIPhone` to be a part of the XCTest extension but the `isIpad` to not be. I understand the reason one at the moment is only used the Screen object while another is used in the test itself. I always found it better to be consistent where we'll be checking the interfaces though, making a conscious decision to leave those checks in the page object or in the tests themselves, personal preference is in the page objects to keep that kind of implementation detail at a lower level. And right now we basically use it in the test to increase the number of words we use which we'll eventually abstract away. I think we could just move the `isIPhone` out of the extension and that'll still keep everything passing and allow us to use it both places or we could move it into BaseScreen but then we'd have to find a better way for the long title tests which isn't really important right now.

Reply:
Good catch, I agree. :) I already have a note to revisit the `isIphone` check as part of my next work on the publishing flow test so there's more consistency. I almost fixed it in the last PR but didn't want to get too sidetracked into work on the editor tests.